File: app/app/src/main/java/williamhester/me/breadit2/ui/text/SpoilerSpan.java
package williamhester.me.breadit2.ui.text;

import android.support.annotation.NonNull;
import android.text.TextPaint;
import android.text.style.ClickableSpan;
import android.view.View;

/**
 * Created by william on 7/20/14.
 */
public class SpoilerSpan extends ClickableSpan {

  private boolean clicked = false;

  @Override
  public void onClick(View view) {
    TextPaint unpaint = new TextPaint();
    clicked = true;
    updateDrawState(unpaint);
    view.invalidate();
  }

  @Override
  public void updateDrawState(@NonNull TextPaint ds) {
    ds.setUnderlineText(true);
    if (!clicked) {
      ds.bgColor = ds.getColor();
    } else {
      ds.bgColor = 0;
    }
  }
}

File: app/app/src/main/java/me/williamhester/reddit/ui/text/SpoilerSpan.kt
package me.williamhester.reddit.ui.text

import android.text.TextPaint
import android.text.style.ClickableSpan
import android.view.View

/** A [ClickableSpan] that hides the text until it is clicked. */
class SpoilerSpan : ClickableSpan() {

  private var clicked = false

  override fun onClick(view: View) {
    val unpaint = TextPaint()
    clicked = true
    updateDrawState(unpaint)
    view.invalidate()
  }

  override fun updateDrawState(ds: TextPaint) {
    ds.isUnderlineText = true
    if (!clicked) {
      ds.bgColor = ds.color
    } else {
      ds.bgColor = 0
    }
  }
}
